app/controllers/concerns/two_factor_authenticatable_methods.rb

```diff
@@ -17,7 +17,15 @@ def handle_valid_verification_for_authentication_context(auth_method:)
     if IdentityConfig.store.feature_new_device_alert_aggregation_enabled &&
        user_session[:new_device] != false
       if current_user.sign_in_new_device_at.blank?
-        current_user.update(sign_in_new_device_at: disavowal_event.created_at)
+        if current_user.events.where(event_type: 'sign_in_notification_timeframe_expired').present?
+          current_user.update(
+            sign_in_new_device_at: current_user.events.where(
+              event_type: 'sign_in_notification_timeframe_expired',
+            ).order(created_at: :desc).limit(2).pluck(:created_at).last,
```

Comment:
Does this cause two separate database queries for retrieving the event? Can we limit this to one?

Reply:
Yes! Thanks for that I have refactored the query.